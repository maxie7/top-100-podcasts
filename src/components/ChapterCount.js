import React from 'react';
import PropTypes from 'prop-types';

const ChapterCount = ({ countChapters }) => {
  return <h2 className="podcast__counter">Episodes: {countChapters} </h2>;
};

ChapterCount.propTypes = {
  countChapters: PropTypes.number
};

export default ChapterCount;
