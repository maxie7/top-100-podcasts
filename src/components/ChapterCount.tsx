import React from 'react';

type ChapterCountProps = {
  countChapters: number
};

const ChapterCount = ({ countChapters }: ChapterCountProps) => (
    <h2 className="podcast__counter">Episodes: {countChapters}</h2>
);

export default ChapterCount;
