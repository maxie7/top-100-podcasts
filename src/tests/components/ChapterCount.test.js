import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ChapterCount from '../../components/ChapterCount';

it('should render ChapterCount section', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ChapterCount countChapters={10} />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('h2');
  expect(renderer.render(<ChapterCount />)).toMatchSnapshot();
  renderer.unmount();
});
