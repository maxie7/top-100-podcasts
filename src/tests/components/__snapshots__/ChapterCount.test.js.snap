// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChapterCount section 1`] = `
<h2
  className="podcast__counter"
>
  Episodes: 
</h2>
`;
